test/uk/gov/hmrc/helptosavestub/controllers/SquidControllerSpec.scala

```diff
@@ -0,0 +1,983 @@
+  "Squid Controller" must {
+
+    "return 415 if the mime type is not application/json" in {
+      val fakeRequestWithoutJson = FakeRequest("POST", "/help-to-save-stub/create-account")
+      val result = new SquidController(messagesApi).createAccount()(fakeRequestWithoutJson)
+      status(result) shouldBe 415
+    }
+
+    "return 200 when requested" in {
+      val result = new SquidController(messagesApi).createAccount()(fakeRequest)
+      status(result) shouldBe 200
+    }
+
+    "The request should return some content" in {
+      //Note that if this is returning a blank page, the controller configuration in application.conf is broken
+      val result = new SquidController(messagesApi).createAccount()(fakeRequest)
+      status(result)
+      val len = result.body.contentLength.getOrElse(0L)
+      len shouldBe 0
+    }
+
+    "If the stub is sent a request with no JSON content it should return a 400" in {
+      val fakeRequestWithoutJson = FakeRequest("POST", "/help-to-save-stub/create-account").withHeaders((CONTENT_TYPE, "application/json"))
+      val result = new SquidController(messagesApi).createAccount()(fakeRequestWithoutJson)
+      status(result) shouldBe 400
+    }
+
+    "If the stub is sent a request with no JSON content it should have an Error object in the response with the errorMessageId set as AAAA0002" in {
+      val fakeRequestWithoutJson = FakeRequest("POST", "/help-to-save-stub/create-account").withHeaders((CONTENT_TYPE, "application/json"))
+      val result: Future[Result] = new SquidController(messagesApi).createAccount()(fakeRequestWithoutJson)
+      status(result)
+      val json: JsValue = contentAsJson(result)
+      val errorMessageId = (json \ "error" \ "errorMessageId").get.asOpt[String]
+      errorMessageId shouldBe Some(NO_JSON_ERROR_CODE)
+    }
+
+    "If the stub is sent a request with no JSON content it should have an Error object in the response with the " +
+      "error message set to message site.no-json" in {
+      val fakeRequestWithoutJson = FakeRequest("POST", "/help-to-save-stub/create-account").withHeaders((CONTENT_TYPE, "application/json"))
+      val result: Future[Result] = new SquidController(messagesApi).createAccount()(fakeRequestWithoutJson)
+      status(result)
+      val json: JsValue = contentAsJson(result)
+      val errorMessage = (json \ "error" \ "errorMessage").get.asOpt[String]
+      assert(messagesApi.isDefinedAt("site.no-json"))
+      errorMessage.getOrElse("") shouldBe messagesApi("site.no-json")
+    }
+
+    "If the stub is sent a request with no JSON content it should have an Error object in the response with the " +
+      "error detail set to message site.no-json-detail" in {
+      val fakeRequestWithoutJson: FakeRequest[AnyContentAsEmpty.type] = FakeRequest("POST", "/help-to-save-stub/create-account").withHeaders((CONTENT_TYPE, "application/json"))
+      val result: Future[Result] = new SquidController(messagesApi).createAccount()(fakeRequestWithoutJson)
```

Comment:
Do you need a new `SquidController` in every test - you can instantiate one at the start of the file and reuse it in each test